File: app/src/main/java/pl/piotrskiba/dailywallpaper/asynctasks/GetBitmapFromUrlAsyncTask.java
package pl.piotrskiba.dailywallpaper.asynctasks;

import android.graphics.Bitmap;
import android.os.AsyncTask;

import pl.piotrskiba.dailywallpaper.interfaces.BitmapLoadedListener;
import pl.piotrskiba.dailywallpaper.utils.NetworkUtils;

public class GetBitmapFromUrlAsyncTask extends AsyncTask<String, Void, Bitmap> {

    private BitmapLoadedListener listener;

    public GetBitmapFromUrlAsyncTask(BitmapLoadedListener listener){
        this.listener = listener;
    }

    @Override
    protected Bitmap doInBackground(String... strings) {
        String url = strings[0];
        return NetworkUtils.getBitmapFromURL(url);
    }

    @Override
    protected void onPostExecute(Bitmap bitmap) {
        listener.onBitmapLoaded(bitmap);
        super.onPostExecute(bitmap);
    }
}

File: app/src/main/java/pl/piotrskiba/dailywallpaper/asynctasks/GetBitmapFromUrlAsyncTask.kt
package pl.piotrskiba.dailywallpaper.asynctasks

import android.graphics.Bitmap
import android.os.AsyncTask
import pl.piotrskiba.dailywallpaper.interfaces.BitmapLoadedListener
import pl.piotrskiba.dailywallpaper.utils.NetworkUtils.getBitmapFromURL

class GetBitmapFromUrlAsyncTask(private val listener: BitmapLoadedListener) : AsyncTask<String, Void, Bitmap>() {
    override fun doInBackground(vararg strings: String?): Bitmap {
        val url = strings[0]!!
        return getBitmapFromURL(url)!!
    }

    override fun onPostExecute(bitmap: Bitmap?) {
        listener.onBitmapLoaded(bitmap!!)
        super.onPostExecute(bitmap)
    }

}